source/server-discovery-and-monitoring/tests/README.rst

```diff
@@ -262,6 +262,38 @@ Run the following test(s) on MongoDB 4.4+.
              mode: "off",
          });
 
+Heartbeat Tests
+~~~~~~~~~~~~~~~
+
+1. Test that ``ServerHeartbeatStartedEvent`` is emitted after the monitoring socket was created and before the ``hello`` is sent to the server
+
+   #. Create a mock TCP server (example shown below) that pushes a ``client connected`` event to a shared array when a client connects and a ``client hello received`` event when the server receives the client hello and then closes the connection::
+         
+        let events = [];
+        server = createServer(clientSocket => {
+          events.push('client connected');
+
+          clientSocket.on('data', () => {
+            events.push('client hello received');
+            clientSocket.end();
+          });
+        });
+        server.listen(9999);
+
+   #. Create a client with ``serverSelectionTimeoutMS: 500, maxPoolSize: 1, minPoolSize: 0`` and listen to ``ServerHeartbeatStartedEvent`` and ``ServerHeartbeatFailedEvent``, pushing the event name to the same shared array as the mock TCP server.
+
+   #. Attempt to connect client to previously created TCP server, catching the error when the client fails to connect 
+
+   #. Assert that the event array has a length of 4
+
+   #. Assert that the first event is the ``client connected`` event
+
+   #. Assert that the second event is the ``serverHeartbeatStartedEvent``
+
+   #. Assert that the third event is the ``client hello received`` event
+
+   #. Assert that the fourth event is the ``serverHeartbeatFailedEvent``
```

Comment:
What's the significance of the "failed" event?  It seems like we could just

- test against a standalone
- connect
- confirm the ordering of events (connected, heartbeat started, hello received, heartbeat succeeded)

Reply:
The `failed` event is expected as the mock server doesn't need to be able to properly reply to the hello for this test to prove the ordering so we expect that it is the case that it won't. As a result, we just want to be sure that the driver fails its heartbeat since the monitoring connection will be closed. I'm open to removing that as a requirement for the test, but it seems appropriate to include.

When you say test against a standalone, how would we track when the connection was created and the hello was received when testing against an actual Mongo server?